src/components/OnBoarding/OnBoarding.jsx

```diff
@@ -0,0 +1,223 @@
+    <div className={styles.chevronRight}>
+      <ChevronRight strokeWidth={2} size={22} />
+    </div>
+  </div>
+)
+Next.propTypes = {
+  onNext: PropTypes.func,
+  lastStep: PropTypes.bool
+}
+
+const Slide = ({ curStep, totalSteps, title, onNext, onPrevious, lastStep, image, text }) => {
+  const firstStep = curStep === 0 || totalSteps === 1
+  return (
+    <div className={styles.slide}>
+
+      {<Previous firstStep={firstStep} onPrevious={onPrevious} />}
+      <div className={styles.Rectangle} />
+      {<Next lastStep={lastStep} onNext={onNext} />}
+    </div>
+  )
+}
+Slide.propTypes = {
+  title: PropTypes.string.isRequired,
+  image: PropTypes.string,
+  text: PropTypes.string,
+  lastStep: PropTypes.bool,
+  onNext: PropTypes.func,
+  onPrevious: PropTypes.func,
+  curStep: PropTypes.number,
+  totalSteps: PropTypes.number
+
+}
+
+const OnBoarding = ({ curStep, totalSteps, onClose, width, height, padding, children, onNext, onPrevious, slideContent }) => {
+  const { title, image, text } = slideContent
+  const lastStep = curStep === totalSteps - 1 || totalSteps === 1
+  return (
+    <div className={styles.OnBoarding}>
+      <div className={styles.overlay} />
+      <div
+        className={styles.onBoarding}
+        style={{
+          width,
+          height
+        }}
+      >
+        <Space size={41} />
+        <div className={styles.title}>
+          <Typography type='title-onboarding'>{title} </Typography>
+        </div>
+        <Space size={40} />
+        <Slide
+          title={title}
+          image={image}
+          text={text}
+          onNext={onNext}
+          onPrevious={onPrevious}
+          lastStep={lastStep}
+          curStep={curStep}
+          totalSteps={totalSteps}
+        />
+        <Space size={23} />
+        <div className={styles.text}>
+          <Typography type='paragraph-onboarding'>{text} </Typography>
+        </div>
+        <Space size={23} />
+        {(!lastStep) && <Bullets curStep={curStep} totalSteps={totalSteps} />}
+        {onClose && <X className={styles.close} size={25} onClick={onClose} />}
+        {children}
+        <div className={styles.finishButon}>
+          {lastStep && <Button classeName={styles.finish} color='primary' onClick={onClose}>Ok, j'ai compris</Button>}
+        </div>
+
+      </div>
+    </div>
+  )
+}
+class ResponsiveOnBoarding extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      step: 0,
+      firstStep: true
+    }
+    this.handleNext = this.handleNext.bind(this)
+    this.handlePrevious = this.handlePrevious.bind(this)
+  }
+
+  handleNext () {
+    this.setState({
+      step: this.state.step += 1
+    })
+  }
+
+  handlePrevious () {
+    this.setState({
+      step: this.state.step -= 1
+    })
+  }
+
+  render () {
+    const { width, height, onClose, children, padding, slidesContent } = this.props
+
+    return (
+      <div>
+        <MediaQuery maxWidth={styles.mobileWidth}>
```

Comment:
I think it's time to use the hook, https://github.com/contra/react-responsive#with-hooks, just retrieve a boolean to know if we are on a desktop or mobile screen, and set props accordingly to OnBoarding